extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/HttpServerMetricsTagsContributor.java

```diff
@@ -23,5 +23,14 @@ interface Context {
         HttpServerRequest request();
 
         HttpResponse response();
+
+        /**
+         * Gives access to the contextual data that was added while the HTTP request was active.
+         * This can be found of doing something like {@link io.smallrye.common.vertx.ContextLocals#get(String)},
+         * however this method is needed because {@link io.smallrye.common.vertx.ContextLocals#get(String)} won't
+         * work when {@link HttpServerMetricsTagsContributor#contribute(Context)} is called as the HTTP request has
+         * already gone away.
+         */
+        Object requestContextLocalData(Object key);
```

Comment:
Would it make sense to add some inferring here to avoid casts? 

```suggestion
        <T> T requestContextLocalData(Object key);
```

Then one could use: `String value = requestLocalData("foo")`

Reply:
There is no type to use for inference

Reply:
I'm talking about the expected return type, as a developer it prevents ugly casts in the code as it can be done in the implementation

Reply:
Yeah I know. But that would mean having another argument that captures the type, which I don't think is worth it.

Reply:
Well, not really, you can simply cast to `(T)` and it should just work ™ :)

```java
        @Override
        public <T> T requestContextLocalData(Object key) {
            return (T) ((HttpServerRequestInternal) request).context().getLocal(key);
        }
```

Reply:
Feel free to push a commit with that

Reply:
Done in [`c9509a5` (#47887)](https://github.com/quarkusio/quarkus/pull/47887/commits/c9509a5ffe7143966c6df86dd1c987fa551b232c)